### PR TITLE
Add io.github.ciromattia.kcc

### PR DESCRIPTION
**Pull request by catsout**
### Please confirm your submission meets all the criteria
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*  
      ~~`filesystem=host`, as the books may be anywhere.~~

- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** [Flatpak support](https://github.com/ciromattia/kcc/issues/383)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
       ~~some pathes for p7zip latestly released at 2016.~~

Note: 
~~I have no idea about how to include kindlegen.~~  Use qemu-i386 to run it, kindlegen should be at `${XDG_DATA_HOME}/kindlegen`.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by catsout** [comment]
bot, build com.github.KCC

**Comment by catsout** [comment]
bot, build com.github.KCC

**Comment by nedrichards** [comment]
bot, build com.github.KCC

**Comment by catsout** [comment]
It seems that the Qt of org.kde.Sdk is build against opengl es on aarch64.  
But PyQt5 is using some desktop OpenGL specific code. [link](https://bugs.debian.org/cgi-bin/bugreport.cgi?att=0;bug=948112;msg=10)  
Try configure disable-feature

**Comment by catsout** [comment]
bot, build com.github.KCC

File: com.github.KCC.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+modules:
+  - name: p7zip
+    no-autogen: true
+    make-args:
+      - 7z
+    sources:
+      - type: archive
+        url: https://sourceforge.net/projects/p7zip/files/p7zip/16.02/p7zip_16.02_src_all.tar.bz2
```
Comment (barthalion) [inline com.github.KCC.yml:20]: Should be upgraded to 17.03 and I guess most of these patches dropped too.

File: com.github.KCC.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
+app-id: com.github.KCC
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: kcc
+finish-args:
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=host
```
Comment (barthalion) [inline com.github.KCC.yml:10]:
```suggestion
  - --filesystem=home
```

File: com.github.KCC.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline com.github.KCC.metainfo.xml:1]: Should be upstreamed.

File: com.github.KCC.desktop [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
Comment (barthalion) [inline com.github.KCC.desktop:1]: Should be upstreamed.

File: com.github.KCC.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+app-id: com.github.KCC
```
Comment (barthalion) [inline com.github.KCC.yml:1]: This is not an app by GitHub, so it should io.github.ciromattia.kcc.

File: com.github.KCC.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+sdk: org.kde.Sdk
+command: kcc
+finish-args:
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --filesystem=host
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+modules:
```
Comment (barthalion) [inline com.github.KCC.yml:13]: The file should be reformatted to follow https://github.com/flathub/flathub/wiki/YAML-Style-Guide

**Comment by catsout** [comment]
bot, build com.github.KCC

**Comment by catsout** [comment]
bot, build io.github.ciromattia.kcc

**Comment by barthalion** [comment]
cc @AcidWeb

I know the old PR went stagnant but how about proceeding with this one?

**Comment by AcidWeb** [comment]
I would love flatpack support but I currently don't have any spare time for KCC project.

**Comment by barthalion** [comment]
Yeah, I guess it's fine; just wanted to make sure you're still okay with having it on Flathub.

/merge @AcidWeb

**Pull request closed without merging**